File: backend/src/main/java/com/clothesshop/backend/entities/ItemInfo.java
package com.clothesshop.backend.entities;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import lombok.Data;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.UpdateTimestamp;


@Entity
@Data
@DynamicUpdate
public class ItemInfo implements Serializable {

  @Id
  private String itemId;

  @NotNull
  private String itemName;

  @NotNull
  private BigDecimal itemPrice;

  @NotNull
  @Min(0)
  private Integer itemStock;

  private String itemDescription;

  private String itemIcon;

  /**
   * 0: on-sale 1: off-sale
   */

  @ColumnDefault("0")
  private Integer itemStatus;

  @ColumnDefault("0")
  private Integer categoryType;

  @CreationTimestamp
  private Date createTime;
  @UpdateTimestamp
  private Date updateTime;

  public ItemInfo() {
  }
}

File: frontend/app/src/app/models/itemInfo.ts
import {ItemInOrder} from './ItemInOrder';

export class ItemInfo {
    itemId: string;
    itemName: string;
    itemPrice: number;
    itemStock: number;
    itemDescription: string;
    itemIcon: string;
    itemStatus: number; // 0: onsale 1: offsale
    categoryType: number;
    createTime: string;
    updateTime: string;


    constructor(itemInOrder?: ItemInOrder) {
        if (itemInOrder) {
            this.itemId = itemInOrder.itemId;
            this.itemName = itemInOrder.itemName;
            this.itemPrice = itemInOrder.itemPrice;
            this.itemStock = itemInOrder.itemStock;
            this.itemDescription = itemInOrder.itemDescription;
            this.itemIcon = itemInOrder.itemIcon;
            this.categoryType = itemInOrder.categoryType;
            this.itemStatus = 0;
        } else {
            this.itemId = '';
            this.itemName = '';
            this.itemPrice = 20;
            this.itemStock = 100;
            this.itemDescription = '';
            this.itemIcon = '';
            this.categoryType = 0;
            this.itemStatus = 0;
        }
    }
}

